src/screens/CreatePost/index.js

```diff
@@ -492,26 +495,31 @@ const CreatePost = () => {
               params: {refresh: true}
             });
           }
-
-          Analytics.logEvent('create_post', {
-            id: 6,
-            newpost_reach: renderLocationString(geoList[geoSelect]),
-            newpost_privacy: listPrivacy[0].label,
-            num_images: 0,
-            added_poll: isPollShown,
-            topics_added: listTopic,
-            anon: typeUser,
-            predicted_audience: audienceEstimations
+        } else {
+          showMessage({
+            message: StringConstant.createPostFailedGeneralError,
+            type: 'danger'
           });
+        }
 
-          return true;
+        Analytics.logEvent('create_post', {
+          id: 6,
+          newpost_reach: renderLocationString(geoList[geoSelect]),
+          newpost_privacy: listPrivacy[0].label,
+          num_images: 0,
+          added_poll: isPollShown,
+          topics_added: listTopic,
+          anon: typeUser,
+          predicted_audience: audienceEstimations
+        });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This part of the code logs an event with various details about the post. It's important to ensure that this logging happens only after successful post creation. In the current implementation, this log event will be sent even if the post creation fails. Consider moving this block inside the success condition (`if (post.code === 200) { ... }`) to ensure accurate analytics.

```diff
+        if (post.code === 200) {
+          showMessage({
+            message: StringConstant.createPostDone,
+            type: 'success'
+          });
 
+          if (isInCreatePostTopicScreen) {
+            navigateToTopicPage();
+          }
+
+          Analytics.logEvent('create_post', {
+            id: 6,
+            newpost_reach: renderLocationString(geoList[geoSelect]),
+            newpost_privacy: listPrivacy[0].label,
+            num_images: 0,
+            added_poll: isPollShown,
+            topics_added: listTopic,
+            anon: typeUser,
+            predicted_audience: audienceEstimations
+          });
+        } else {
+          showMessage({
+            message: StringConstant.createPostFailedGeneralError,
+            type: 'danger'
+          });
+        }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->